tonic/src/transport/server/mod.rs

```diff
@@ -221,6 +226,30 @@ impl Server {
         Router::new(self.clone(), svc)
     }
 
+    /// Create a router with the optional `S` typed service as the first service.
+    ///
+    /// This will clone the `Server` builder and create a router that will
+    /// route around different services.
+    pub fn add_optional_service<S>(
+        &mut self,
+        svc: Option<S>,
+    ) -> Router<Either<S, EmptyService>, Unimplemented>
+    where
+        S: Service<Request<Body>, Response = Response<BoxBody>>
+            + NamedService
+            + Clone
+            + Send
+            + 'static,
+        S::Future: Send + 'static,
+        S::Error: Into<crate::Error> + Send,
+    {
+        let svc = match svc {
+            Some(some) => Either::A(some),
+            None => Either::B(EmptyService),
```

Comment:
I guess I missed this before but this is where my brain was breaking. Is there a reason we can't reuse the `Unimplemented` service here?

Reply:
I was unsure and was hoping you could tell me. If you see no problem with it then I'll go ahead.